### ISSUE TITLE
[Documentation needed] [Android] Crash when showing a dialog while app is paused

### ISSUE THREAD

**Issue by SimonSimCity** [aritchie/userdialogs#305]
## Please check all of the platforms you are having the issue on (if platform is not listed, it is not supported)

 - [ ] iOS
 - [x] Android
 - [ ] UWP
 - [ ] PCL

## Expected Behaviour

See `What I would this library expect to do:`

## Actual Behavior

App crashes.

## Details

As noted in #230 and #262, the app crashes on Android if the user tries to show a dialog after the activity is paused or stopped. This is because both `Show()` implementations on `UserDialogsImpl` use `AbstractAppCompatDialogFragment.Show()` or `AbstractDialogFragment.Show()` - see https://github.com/aritchie/userdialogs/blob/master/src/Acr.UserDialogs.Android/UserDialogsImpl.cs#L306. If you take a look at the `Show()` method, it uses `FragmentTransaction.Commit()`, which doesn't allow the transaction to be executed if the view isn't visible to the user.

As this isn't a quite uncommon situation (f.e. showing a message if a network-task times out could easily happen after the user switched to another app), I would expect this library to link to a documentation, where I can read about a best practice to catch this.

The easiest way,would be to use an implementation that uses `FragmentTransaction.CommitAllowingStateLoss()` - See http://stackoverflow.com/questions/12105064/actions-in-onactivityresult-and-error-can-not-perform-this-action-after-onsavei/12464899#12464899 - but this is actually a bad practice unless you know exactly what you're doing. But there are some situations where it is wanted. Said that, this library should provide a way to do it, but make the developer aware of that this dialog may not be restored when restoring the UI. Most of the other situations I found suffer of the same problem - See http://stackoverflow.com/questions/15729138/on-showing-dialog-i-get-can-not-perform-this-action-after-onsaveinstancestate.

What I expect this library to do:
* Make the user aware of that there is a potential risk to get to this exception when opening a user dialog asynchronously. Advise them to avoid opening user dialogs in async methods where ever possible.
* Open a way the user could show a message, that's (kind of) not that important - saying, that this message may will get lost if the app is re-initialized after being killed.

A post worth reading in order to get a better understanding:

http://www.androiddesignpatterns.com/2013/08/fragment-transaction-commit-state-loss.html

**Comment by aritchie** [comment]
This library is not going to deal with backgrounding issues.  It opens dialogs according to what is requested. 

If there is a feature or docs you expect, you are welcome to contribute

**Comment by fwermelinger** [comment]
@SimonSimCity did you end up creating a PR for those things? We ran into some issues with this while developing our app and would certainly appreciate it.

**Comment by SimonSimCity** [comment]
@fwermelinger What would you expect to be a good solution? To drop the dialog or to show it as soon as the application is available again? Both have valid scenarios.

**Comment by fwermelinger** [comment]
@SimonSimCity I think adding an option to allow dropping the dialog would be good. 

And additionally give a more precise exception or warning message if this happens. Could you throw a custom exception if a dialog is opened while the app is in the background and state-loss is not allowed?

**Comment by SimonSimCity** [comment]
What I did in my project was pretty simple ... I've just implemented a custom error-handler by wrapping another class around the actual implementation of `IUserClass` which I now use. By doing this, the application does not crash anymore when an exception is received but it is logged and swallowed as all our other exceptions are. This is kind of working but feels like going against the principle of crash-only-software (which IMO is the best principle for writing apps) because we do not let the "ill" part of the application crash and restart, but it gets us on in the way we decided to go.

A good reading on this: https://en.m.wikipedia.org/wiki/Crash-only_software
